internal/app/machined/pkg/controllers/k8s/control_plane_static_pod.go

```diff
@@ -328,33 +359,51 @@ func (ctrl *ControlPlaneStaticPodController) manageControllerManager(ctx context
 
 	args := []string{
 		"/usr/local/bin/kube-controller-manager",
-		"--allocate-node-cidrs=true",
-		"--bind-address=127.0.0.1",
-		"--port=0",
-		fmt.Sprintf("--cluster-cidr=%s", strings.Join(cfg.PodCIDRs, ",")),
-		fmt.Sprintf("--service-cluster-ip-range=%s", strings.Join(cfg.ServiceCIDRs, ",")),
-		fmt.Sprintf("--cluster-signing-cert-file=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "ca.crt")),
-		fmt.Sprintf("--cluster-signing-key-file=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "ca.key")),
-		"--controllers=*,tokencleaner",
-		"--configure-cloud-routes=false",
-		fmt.Sprintf("--kubeconfig=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "kubeconfig")),
-		fmt.Sprintf("--authentication-kubeconfig=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "kubeconfig")),
-		fmt.Sprintf("--authorization-kubeconfig=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "kubeconfig")),
-		"--leader-elect=true",
-		fmt.Sprintf("--root-ca-file=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "ca.crt")),
-		fmt.Sprintf("--service-account-private-key-file=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "service-account.key")),
-		"--profiling=false",
 		"--use-service-account-credentials",
 	}
 
+	builder := argsbuilder.Args{
+		"allocate-node-cidrs":              "true",
+		"bind-address":                     "127.0.0.1",
+		"port":                             "0",
+		"cluster-cidr":                     strings.Join(cfg.PodCIDRs, ","),
+		"service-cluster-ip-range":         strings.Join(cfg.ServiceCIDRs, ","),
+		"cluster-signing-cert-file":        filepath.Join(constants.KubernetesControllerManagerSecretsDir, "ca.crt"),
+		"cluster-signing-key-file":         filepath.Join(constants.KubernetesControllerManagerSecretsDir, "ca.key"),
+		"controllers":                      "*,tokencleaner",
+		"configure-cloud-routes":           "false",
+		"kubeconfig":                       filepath.Join(constants.KubernetesControllerManagerSecretsDir, "kubeconfig"),
+		"authentication-kubeconfig":        filepath.Join(constants.KubernetesControllerManagerSecretsDir, "kubeconfig"),
+		"authorization-kubeconfig":         filepath.Join(constants.KubernetesControllerManagerSecretsDir, "kubeconfig"),
+		"leader-elect":                     "true",
+		"root-ca-file":                     filepath.Join(constants.KubernetesControllerManagerSecretsDir, "ca.crt"),
+		"service-account-private-key-file": filepath.Join(constants.KubernetesControllerManagerSecretsDir, "service-account.key"),
+		"profiling":                        "false",
+	}
+
 	if cfg.CloudProvider != "" {
-		args = append(args, fmt.Sprintf("--cloud-provider=%s", cfg.CloudProvider))
+		builder.Set("cloud-provider", cfg.CloudProvider)
+	}
+
+	mergePolicies := argsbuilder.MergePolicies{
+		"cluster-cidr":             argsbuilder.MergeAdditive,
```

Comment:
I feel that `cluster-cidr` and `service-cluster-ip-range` should be overridable completely; if someone needs to override them, let them do it; additive policy doesn't allow to remove our default values